src/pybind/rados/rados.pyx

```diff
@@ -2536,6 +2539,80 @@ cdef class Ioctx(object):
             raise make_ex(ret, "error reading %s" % object_name)
         return completion
 
+    @requires(('object_name', str_type), ('xattr_name', str_type),
+              ('xattr_value', bytes), ('oncomplete', opt(Callable)))
+    def aio_set_xattr(self, object_name, xattr_name, xattr_value, oncomplete = None):
+        """
+        Asynchronously set an extended attribute on an object
+
+        :param object_name: the name of the object to set xattr to
+        :type object_name: str
+        :param xattr_name: which extended attribute to set
+        :type xattr_name: str
+        :param xattr_value: the value of the  extended attribute
+        :type xattr_value: bytes
+        :param oncomplete: what to do when the setxttr completes
+        :type oncomplete: completion
+
+        :raises: :class:`Error`
+        :returns: completion object
+        """
+        object_name = cstr(object_name, 'object_name')
+        xattr_name = cstr(xattr_name , 'xattr_name')
+
+        cdef:
+            Completion completion
+            char* _object_name = object_name
+            char* _xattr_name = xattr_name
+            char* _xattr_value = xattr_value
+            size_t xattr_value_len = len(xattr_value)
+
+        completion = self.__get_completion(oncomplete, None)
+        self.__track_completion(completion)
+        with nogil:
+            ret = rados_aio_setxattr(self.io, _object_name,
+                               completion.rados_comp,
+                               _xattr_name, _xattr_value, xattr_value_len)
+
+        if ret < 0:
+            completion._cleanup()
+            raise make_ex(ret, "Failed to set xattr %r" % xattr_name)
+        return completion
+
+    @requires(('object_name', str_type), ('xattr_name', str_type), ('oncomplete', opt(Callable)))
+    def aio_rm_xattr(self, object_name, xattr_name, oncomplete = None):
+        """
+        synchronously delete an extended attribute from an object
```

Comment:
should be `s/synchronously/Asynchronously`

Reply:
@ideepika updated, please reveiw.

Reply:
updated, please review, thanks a lot.

Reply:
test code:
```
from rados import (Rados, WriteOpCtx, LIBRADOS_CREATE_EXCLUSIVE, NoData)
from nose.tools import eq_ as eq, ok_ as ok, assert_raises
import threading
import rados
import unittest

class TestIoctx(unittest.TestCase):
    def setUp(self):
        self.rados = Rados(conffile='./ceph.conf')
        self.rados.connect()
        self.rados.create_pool('pool002')
        assert self.rados.pool_exists('pool002')
        self.ioctx = self.rados.open_ioctx('pool002')
    def tearDown(self):
        self.ioctx.close()
        self.rados.delete_pool('pool002')
        self.rados.shutdown()

    def test_aio_set_xattr(self):
        lock = threading.Condition()
        count = [0]
        def cb(blah):
            with lock:
                count[0] += 1
                lock.notify()
            return 0
        comp = self.ioctx.aio_set_xattr("obj", "key", "value", cb)
        comp.wait_for_complete()
        with lock:
            while count[0] < 1:
                lock.wait()
        eq(comp.get_return_value(), 0)
        eq(self.ioctx.get_xattr('obj', 'key'), 'value')

    def test_aio_rm_xattr(self):
        lock = threading.Condition()
        count = [0]
        def cb(blah):
            with lock:
                count[0] += 1
                lock.notify()
            return 0
        self.ioctx.set_xattr('xyz', 'key', 'value')
        eq(self.ioctx.get_xattr('xyz', 'key'), 'value')
        comp = self.ioctx.aio_rm_xattr('xyz', "key", cb)
        comp.wait_for_complete()
        with lock:
            while count[0] < 1:
                lock.wait()
        eq(comp.get_return_value(), 0)
        with assert_raises(NoData):
            self.ioctx.get_xattr('xyz', 'key')

if __name__ == '__main__':
    unittest.main()
```
test results:
```
[root@jiaowa build]# python test_aio_xttrs.py
..
----------------------------------------------------------------------
Ran 2 tests in 5.792s

OK
```
please check.